### ISSUE TITLE
[Question] What is the plan with regards to handling bindings for multiple c++ classes in the same project

### ISSUE THREAD

**Issue by zicklag** [ncannasse/webidl#14]
@zicklag ... while I have you here maybe you can answer something for me?  What is the plan with regards to handling bindings for multiple c++ classes in the same project.  The idl file in the sample is called point.idl presumably because the only thing it handles presently is the Point class but if I wanted to  add a class (different header and implementation file) would I just append it to the same .idl file?  And if so I  suspect I will have to handle it somehow in the makefile?  Perhaps make is a bit light for a larger workload if that's the case? Perhaps the work presently being done by make should be handled by a webidl executable?  I'm not sure what the long term goals are for this project but it looks like something that I would like to make more industrial use of than just a few samples...

_Originally posted by @WolfieWerewolf in https://github.com/ncannasse/webidl/pull/13#issuecomment-464375774_

**Comment by zicklag** [comment]
> but if I wanted to add a class (different header and implementation file) would I just append it to the same .idl file?

Yes. All of the definitions that you want to go into the Haxe module would go into the same WebIDL file.

> And if so I suspect I will have to handle it somehow in the makefile? Perhaps make is a bit light for a larger workload if that's the case? Perhaps the work presently being done by make should be handled by a webidl executable?

Yes, you would want more automation for that that this library doesn't currently provide. I'm actually using this library to automate C++ bindings for [Kha](https://github.com/Kode/Kha) projects. Kha has its own make system called [Khamake](https://github.com/Kode/khamake/), and I have recently submitted PRs for Khamake and Kha that completely automate the whole binding process.

For any other users of this library, though, a setup for automating the make process for other applications would be useful. What I setup was very specific to Kha, but you could still check it out at https://github.com/armory3d/haxebullet/issues/25#issuecomment-462210471.

**Comment by WolfieWerewolf** [comment]
@zicklag great, thanks.. I will take a look.

**Comment by WolfieWerewolf** [comment]
@zicklag .. I think we discussed this and I believe I've handled this in my PR.  Once you've tested I think this ticket  can be closed.

**Comment by zicklag** [comment]
Sweet, I'll test it as soon as I get the time. :+1:
